src/operator/operator_common.h

```diff
@@ -526,17 +526,46 @@ class OpSignature {
    * and the layout to sign the op.
    */
 
-#if MXNET_USE_MKLDNN == 1
+#if MXNET_USE_MKLDNN == 100
   void AddSign(const mkldnn::memory &mem) {
-    auto desc = mem.get_primitive_desc().desc();
-    hash = hash * 2 + desc.data.format;
-    eles.push_back(desc.data.format);
+    auto desc = mem.get_desc();
+    hash = hash * 2 + desc.data.format_kind;
+    eles.push_back(desc.data.format_kind);
     hash = hash * 2 + desc.data.data_type;
     eles.push_back(desc.data.data_type);
     for (int i = 0; i < desc.data.ndims; i++) {
       hash = hash * 2 + desc.data.dims[i];
       eles.push_back(desc.data.dims[i]);
     }
+    switch (desc.data.format_kind) {
```

Comment:
Hashing mkldnn memory is more heavy than before, so shall we cache the hash result somewhere to reduce the hash overhead?

Reply:
Seems need extra for ( loop: desc.data.ndims) && for(loop: desc.data.format_desc.blocking.inner_nblks), I will do some profile to decide whether hash rslt needs cached

Reply:
OK. Ideally, we can save the hash result into MKLDNNMemory, only computing it when hash result doesn't exist.

Reply:
in tests/python/train/test_conv.py  
fwd cache cost: 2.603
total fwd cost: 505.974

bwd cache cost: 1.798
total bwd cost: 720.74

So maybe not that necessary to optimize this code ?